core/trino-main/src/main/java/io/trino/sql/planner/plan/AggregationNode.java

```diff
@@ -232,7 +232,8 @@ public boolean producesDistinctRows()
     {
         return aggregations.isEmpty() &&
                 !groupingSets.getGroupingKeys().isEmpty() &&
-                outputs.size() == groupingSets.getGroupingKeys().size();
+                groupingSets.getGroupingSetCount() == 1 &&
```

Comment:
nit: consider adding comment why we need to check `groupingSets.getGroupingSetCount() == 1`